File: 이코테/P118_개임개발/P118.c
#include<stdio.h>
int main()
{
    int map[52][52] = {0};
    int moved[52][52] = {0};

    int move[4][2] = { {-1,0} ,  {0,1}, {1,0} , {0,-1}  };

    int n,m,y,x,direction;



    scanf("%d %d",&n, &m);

    scanf("%d %d %d",&y, &x, &direction);


    for(int i = 1 ; i <= n ; i++)
    {
        for(int j = 1 ; j <= m ; j++)
        {
            int temp;
            scanf("%d", &temp);
            map[i][j] = !temp;
            
        }
    }


    int result = 1;
    int check = 0;
    y++;
    x++;
    moved[y][x] = 1;
    while(check < 4)
    {
        direction = (direction+3)%4;
        if( map[ y + move[direction][0] ][ x + move[direction][1] ] == 1  &&  moved[ y + move[direction][0] ][ x + move[direction][1] ] == 0 )
        {
            moved[ y + move[direction][0] ][ x + move[direction][1] ] = 1;
            y += move[direction][0] ;
            x += move[direction][1] ;
            result++;
            check = 0;
        }
        else
        {
            check++;
        }

        if( check == 4 && map[ y - move[direction][0] ][ x - move[direction][1] ] == 1 )
        {
            y -= move[direction][0] ;
            x -= move[direction][1] ;
            direction = (direction+1)%4;
            check = 0;
        }
    }

    printf("%d", result);

}

File: 이코테/P118_개임개발/P118.py
n, m = map(int, input().split())

moved   = [[0] * (m+2) for _ in range(n+2)]

movex = [-1,0,1,0]
movey = [0,1,0,-1]

y, x, direction = map(int, input().split())

field = [ ]
field.append([0]*(m+2))
for i in range (n):
    field.append( [0] + list( map( lambda x: 1-int(x),input().split() ) ) +[0] )
field.append([0]*(m+2))

print(field)

result = 1
count = 0
y += 1
x += 1
moved[y][x] = 1
while count < 4:
    
    direction = ( direction + 3 ) % 4
    if field[y + movey[direction]][x + movex[direction]] == 1 and moved[y + movey[direction]][x + movex[direction]] == 0:
        y +=  movey[direction]
        x +=  movex[direction]
        moved[y][x] = 1
        result+=1
        count = 0
    else:
        count+=1
    if(count == 4 and field[y - movey[direction]][x - movex[direction]] == 1):
        y -= movey[direction]
        x -= movex[direction]
        direction = (direction+1)%4;
        count = 0

print(result)
    